src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -69,21 +69,41 @@ public long getRowCountForScale(int scale)
         }
     }
 
-    private long computeCountUsingStaticScale()
+    private long computeCountUsingStaticScale(float scale)
     {
-        return getRowCountForScale(1);
+        if (scale == Math.round(scale)) {
+            return getRowCountForScale(1);
+        }
+        // Handle decimal scales (not part of spec). Will only have an effect on
+        // decimal scales 0 < x < 1.
```

Comment:
I see. So for example joins with `date_dim`, on the date column, will return fewer rows because the other side of the join won't have as many, if any, matches.

If I revert to the old behaviour for static scaling, the difference in size is roughly 2x. With linear scaling on the static tables the output size is 98 MB (current code in PR) and with regular static scaling the output size is 184 MB (same number of rows regardless of scale). I am leaning towards static scaling as it currently exists in the code, i.e. not linearly scaling it. I'll ask Dain and see if he has any objections.